build.gradle.kts

```diff
@@ -0,0 +1,372 @@
+        classpath("com.adarshr:gradle-test-logger-plugin:1.7.0")
+    }
+}
+
+val ideVersions = IdeVersions(project)
+val ideVersion = ideVersions.resolveShortenedIdeProfileName()
+val toolkitVersion: String by project
+val kotlinVersion: String by project
+val mockitoVersion: String by project
+val mockitoKotlinVersion: String by project
+val assertjVersion: String by project
+val junitVersion: String by project
+val remoteRobotPort: String by project
+val ktlintVersion: String by project
+val remoteRobotVersion: String by project
+val ideaPluginVersion: String by project
+// publishing
+val publishToken: String by project
+val publishChannel: String by project
+
+plugins {
+    java
+    id("de.undercouch.download") version "4.1.1" apply false
+}
+
+group = "software.aws.toolkits"
+// please check changelog generation logic if this format is changed
+version = "$toolkitVersion-$ideVersion".toString()
+
+repositories {
+    maven("https://www.jetbrains.com/intellij-repository/snapshots/")
+}
+
+allprojects {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        jcenter()
+    }
+
+    apply(plugin = "com.adarshr.test-logger")
+    apply(plugin = "java")
+    apply(plugin = "jacoco")
+
+    java.sourceCompatibility = JavaVersion.VERSION_1_8
+    java.targetCompatibility = JavaVersion.VERSION_1_8
+
+    tasks.withType(JavaExec::class.java) {
+        systemProperty("aws.toolkits.enableTelemetry", false)
+    }
+
+    tasks.withType(RunIdeTask::class.java) {
+        val ijExt = project.extensions.getByName("intellij") as IntelliJPluginExtension
+        val alternativeIde = System.getenv("ALTERNATIVE_IDE")
+        if (alternativeIde != null) {
+            if (File(alternativeIde).exists()) {
+                ijExt.alternativeIdePath = System.getenv("ALTERNATIVE_IDE")
+            } else {
+                throw GradleException("ALTERNATIVE_IDE path not found $alternativeIde ${if (alternativeIde.endsWith("/")) "remove the trailing slash" else ""}")
+            }
+        }
+    }
+
+    configurations {
+        runtimeClasspath {
+            exclude(group = "org.slf4j")
+            exclude(group = "org.jetbrains.kotlin")
+            exclude(group = "org.jetbrains.kotlinx")
+            exclude(group = "software.amazon.awssdk", module = "netty-nio-client")
+        }
+    }
+}
+
+// Kotlin plugin seems to be bugging out when there are no kotlin sources
+configure(subprojects.filter { it.name != "telemetry-client" }) {
+    apply(plugin = "kotlin")
+
+    sourceSets {
+        getOrCreate("integrationTest") {
+            java.srcDir("it")
+        }
+    }
+}
+
+subprojects {
+    group = parent!!.group
+    version = parent!!.version
+
+    apply(plugin = "java")
+    apply(plugin = "idea")
+    apply(plugin = "com.adarshr.test-logger")
+
+    sourceSets {
+        main.get().java.srcDirs(findFolders(project, "src", ideVersion))
```

Comment:
this greedily gets it right? is there a better way - when I was last poking about with this it was discouraged to do the greedy get - something like 

```
main {
  java.srcDir(findFolders(project, "src", ideVersion))
}
```

Reply:
Ah yeah, changing all of them to that